src/app/components/user/UserWorkspaces/UserWorkspacesComponent.js

```diff
@@ -0,0 +1,243 @@
+import CreateTeamDialog from '../../team/CreateTeamDialog';
+import { FlashMessageSetterContext } from '../../FlashMessage';
+import { getErrorMessageForRelayModernProblem } from '../../../helpers';
+import { stringHelper } from '../../../customHelpers';
+import styles from '../user.module.css';
+import workspaceStyles from './UserWorkspacesComponent.module.css';
+import MediasLoading from '../../media/MediasLoading';
+
+const updateUserMutation = graphql`
+  mutation UserWorkspacesComponentUpdateUserMutation($input: UpdateUserInput!) {
+    updateUser(input: $input) {
+      me {
+        current_team_id
+      }
+    }
+  }
+`;
+
+const UserWorkspacesComponent = ({
+  currentTeam,
+  numberOfTeams,
+  pageSize,
+  relay,
+  teams,
+  totalCount,
+}) => {
+  const [showCreateTeamDialog, setShowCreateTeamDialog] = React.useState(false);
+  const [cursor, setCursor] = React.useState(0);
+  const [isPaginationLoading, setIsPaginationLoading] = React.useState(false);
+  const setFlashMessage = React.useContext(FlashMessageSetterContext);
+
+  const onFailure = (errors) => {
+    const errorMessage = getErrorMessageForRelayModernProblem(errors) || <FormattedMessage
+      id="switchTeams.error"
+      defaultMessage="Sorry, an error occurred while updating the workspace. Please try again and contact {supportEmail} if the condition persists."
+      description="Error message with instructions on how to contact support"
+      values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }}
+    />;
+    setFlashMessage(errorMessage, 'error');
+  };
+
+  const onSuccess = (team) => {
+    const path = `/${team.slug}/all-items`;
+    browserHistory.push(path);
+  };
+
+  const setCurrentTeam = (team) => {
+    commitMutation(Relay.Store, {
+      mutation: updateUserMutation,
+      variables: {
+        input: {
+          current_team_id: team.dbid,
+        },
+      },
+      onCompleted: ({ error }) => {
+        if (error) {
+          onFailure(error);
+        } else {
+          onSuccess(team);
+        }
+      },
+      onError: onFailure,
+    });
+  };
+
+  const handleCancelCreateTeam = () => {
+    setShowCreateTeamDialog(false);
+  };
+
+  return (
+    <>
+      <SettingsHeader
+        title={
+          <FormattedMessage
+            id="userSettings.workspacesTitle"
+            defaultMessage="Workspaces [{workspacesCount}]"
+            description="Title for user settings area for user workspaces list they are a memeber of"
+            values={{ workspacesCount: numberOfTeams || 0 }}
+          />
+        }
+        actionButton={
+          <ButtonMain
+            theme="brand"
+            size="default"
+            variant="contained"
+            onClick={() => setShowCreateTeamDialog(true)}
+            label={
+              <FormattedMessage id="switchTeams.newTeamLink" defaultMessage="Create" description="Button label for initiating creating a new team workspace" />
```

Comment:
Please fix the namespace to match the component name.

Reply:
I'm pair-reviewing this with Sarah and asked to keep existing string ids to avoid generating retranslations.